Documents/GuidingPrinciples.md

```diff
@@ -13,6 +15,26 @@
 
 7. The SARIF design is focused on expressing results as produced by a tool at a specific point-in-time and current excludes detailed thinking related to results management (associated result work item, false positive evaluation, etc.). These concepts may be addressed by defining or proposing 'profiles' that broaden SARIF's design surface area, contingent on progress with core work.
 
+# Principles related to ease of production and consumption
+
 8. In cases where SARIF permits the same information to be expressed in more than one way, any or all of those representations may be present. This allows producers to persist all the information they have, and it allows consumers to choose the representation they prefer. SARIF requires all the representations to be consistent.
 
-    Example: SARIF allows you to specify the start of a `region` either with the `startLine` and `startColumn` properties, or with the `offset` property. It's ok for both to be present, as long as they're consistent.
\ No newline at end of file
+    Example: SARIF allows you to specify the start of a `region` either with the `startLine` and `startColumn` properties, or with the `offset` property. It's ok for both to be present, as long as they're consistent.
+
+9. In cases where it might help producers to provide valid property values, property names may specify their format, for example, `workItemUris`, `startTimeUtc`, and `instanceGuid`.
+
+10. In cases where the a composite property (an array or an object) can legally be empty, SARIF allows a producer to either to emit an empty array or object, or to omit the property (even though including it would make the log file larger).
+
+# Principles related to compactness and readability

```

Comment:
>Principles [](start = 2, length = 10)

These aren't bad but I think they could be improved. 'readability' as a term here also implies 'searchability' or refers to completeness, and we haven't really covered this aspect. In general, I feel like these guidelines are suffering in that some principles are very high level and other items are very targeted to a specific design approach.

All the compactness/readability issues derive from a principle that I don't see expressed here: SARIF's core design is intended to support emitting a flat list/stream of message, such as what might appears in an error list. ie, run.result is the heart of the design and is intended to be self-sufficient. augmenting this core information, we've chosen to provide supplemental tables to augment core results production for other scenarios (these scenarios include debuggability, which is why you can embed files in the files table, and compliance, which is why we support arbitrary file lists and persisting hash values).